e2e/test-runner/test_cases.go

```diff
@@ -85,13 +85,17 @@ func testSystemdEnabled(t *testing.T) {
 	require.Contains(t, string(out), "degraded", "systemd output should be degraded")
 }
 
-// testSystemdUnits ensures the list of failed units does not regress
-func testSystemdUnits(t *testing.T) {
+// testSystemdUnits ensures the list of failed units does not regress.
+func testSystemdUnits(t *testing.T) { //nolint: thelper
 	if !systemdExpected() {
 		// Enable systemd in config file
-		wslCommandAsUser(context.Background(), "root", "bash", "-c", `printf '\n[boot]\nsystemd=true\n' >> /etc/wsl.conf`)
+		out, err := wslCommandAsUser(context.Background(), "root", "bash", "-c", `printf '\n[boot]\nsystemd=true\n' >> /etc/wsl.conf`).CombinedOutput()
+		require.NoError(t, err, "Failed to enable systemd: %s", out)
 		// Restore config file
-		defer wslCommandAsUser(context.Background(), "root", "bash", "-c", `head -n -2 "/etc/wsl.conf" > "/etc/wsl.conf"`)
+		t.Cleanup(func() {
+			out, err := wslCommandAsUser(context.Background(), "root", "bash", "-c", `head -n -2 "/etc/wsl.conf" > "/etc/wsl.conf"`).CombinedOutput()
```

Comment:
I’m not really confortable with that Cleanup approach. Nothing really ensure that the first two lines are what matters in the future.

Can we rather save the original vanilla file somewhere and restore it?

Reply:
Yeah that makes much more sense. On it.